pkg/virt-api/api.go

```diff
@@ -269,6 +269,21 @@ func (app *virtAPIApp) composeSubresources() {
 			Operation(version.Version + "VNC").
 			Doc("Open a websocket connection to connect to VNC on the specified VirtualMachineInstance."))
 
+		subws.Route(subws.GET(rest.ResourcePath(subresourcesvmiGVR) + rest.SubResourcePath("ssh") + rest.PortPath).
+			To(subresourceApp.SSHRequestHandler).
+			Param(rest.NamespaceParam(subws)).Param(rest.NameParam(subws)).
+			Param(rest.SSHPortParameter(subws)).
+			Operation(version.Version + "SSH").
+			Doc("Open a websocket connection to connect to SSH on the specified VirtualMachineInstance and port."))
+
+		subws.Route(subws.GET(rest.ResourcePath(subresourcesvmiGVR) + rest.SubResourcePath("ssh") + rest.InterfacePath + rest.PortPath).
+			To(subresourceApp.SSHRequestHandler).
+			Param(rest.NamespaceParam(subws)).Param(rest.NameParam(subws)).
+			Param(rest.SSHInterfaceParameter(subws)).
+			Param(rest.SSHPortParameter(subws)).
+			Operation(version.Version + "SSH/withInterfaceName").
+			Doc("Open a websocket connection to connect to SSH on the specified VirtualMachineInstance network interface and port."))
```

Comment:
are we sure we can support interfaces beyond the pod interface?  the pod network is the only place i think we're guaranteed virt-api can contact

Reply:
I'm not sure about this myself. I just saw there are multiple interfaces and just taking the first one seemed to simple.
Willing to remove it if somebody can confirm the assumption.